skrub/_fuzzy_join.py

```diff
@@ -400,39 +436,57 @@ def fuzzy_join(
     main_num_cols = main_table[main_cols].select_dtypes(include="number").columns
     aux_num_cols = aux_table[aux_cols].select_dtypes(include="number").columns
 
+    main_time_cols = main_table[main_cols].select_dtypes(include="datetime").columns
+    aux_time_cols = aux_table[aux_cols].select_dtypes(include="datetime").columns
+
+    main_str_cols = list(set(main_cols) - set(main_num_cols) - set(main_time_cols))
+    aux_str_cols = list(set(aux_cols) - set(aux_num_cols) - set(main_time_cols))
+
     # Check if included columns are numeric:
     any_numeric = len(main_num_cols) != 0
+    # Check if included columns are datetime:
+    any_time = len(main_time_cols) != 0
+    # Check if included columns are datetime:
+    any_str = len(main_str_cols) != 0
     # Check if included columns have mixed types:
-    mixed_types = len(main_cols) > len(main_num_cols)
+    # mixed_types = len(main_cols) > (len(main_num_cols) + len(main_time_cols))
 
     if len(main_cols) == 1 and len(aux_cols) == 1 and any_numeric is False:
         main_cols = main_cols[0]
         aux_cols = aux_cols[0]
 
-    if numerical_match in ["number"] and any_numeric and not mixed_types:
-        main_enc, aux_enc = _numeric_encoding(
-            main_table, main_num_cols, aux_table, aux_num_cols
-        )
-        idx_closest, matching_score = _nearest_matches(main_enc, aux_enc)
-    elif numerical_match in ["number"] and any_numeric and mixed_types:
-        main_num_enc, aux_num_enc = _numeric_encoding(
-            main_table, main_num_cols, aux_table, aux_num_cols
-        )
-        main_str_cols = list(set(main_cols) - set(main_num_cols))
-        aux_str_cols = list(set(aux_cols) - set(aux_num_cols))
-        main_str_enc, aux_str_enc = _string_encoding(
-            main_table,
-            main_str_cols,
-            aux_table,
-            aux_str_cols,
-            encoder=encoder,
-            analyzer=analyzer,
-            ngram_range=ngram_range,
-        )
-        main_enc = hstack((main_num_enc, main_str_enc), format="csr")
-        aux_enc = hstack((aux_num_enc, aux_str_enc), format="csr")
+    if numerical_match in ["auto", "number", "time"]:
+        main_enc = np.zeros((len(main_table.index), 1))
```

Comment:
I think the easiest way to stack elements in this situation is to instantiate a list in the beginning, like:
```python
main_embeddings, aux_embeddings = [], []
```
and then append tables to it.
Then use hstack only at the end:

```python
main_embeddings = hstack(main_embeddings, format="csr")
aux_embeddings = hstack(aux_embeddings, format="csr)
```

I already mentioned it but if we don't have string types, we should be careful with using sparse matrices for dense arrays, because it comes with a significant overhead and might be super slow :)

WDYT?